Use kernel_func_name from aotCompiler

priyaramani opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #66337

Right now, assembly code generated for the a given method from the model is named wrapper or func by default. The function name is then replaced with a proper kernel_func_name after target specific assembly is generated. 
This PR propagates a desired kernel_func_name right from aotCompiler API so that the generated function has the needed name that doesn't need to be replaced later.

Differential Revision: [D31514095](https://our.internmc.facebook.com/intern/diff/D31514095)

priyaramani committed:
Use kernel_func_name from aotCompiler

[ghstack-poisoned]

priyaramani committed:
Update on "Use kernel_func_name from aotCompiler"




[ghstack-poisoned]

navahgar reviewed (commented):
Mostly LGTM. Just a couple of minor comments.

Note that my PR to use the given kernel function name in `LLVMCodeGen` had to be reverted (https://github.com/pytorch/pytorch/pull/66216). So, even with this change, the kernel func name that you pass in from `aotCompile` will not be used, until I reland my change.

priyaramani committed:
Update on "Use kernel_func_name from aotCompiler"


Right now, assembly code generated for the a given method from the model is named wrapper or func by default. The function name is then replaced with a proper kernel_func_name after target specific assembly is generated. 
This PR propagates a desired kernel_func_name right from aotCompiler API so that the generated function has the needed name that doesn't need to be replaced later.

Differential Revision: [D31514095](https://our.internmc.facebook.com/intern/diff/D31514095)

[ghstack-poisoned]

priyaramani committed:
Update on "Use kernel_func_name from aotCompiler"


Right now, assembly code generated for the a given method from the model is named wrapper or func by default. The function name is then replaced with a proper kernel_func_name after target specific assembly is generated. 
This PR propagates a desired kernel_func_name right from aotCompiler API so that the generated function has the needed name that doesn't need to be replaced later.

Differential Revision: [D31514095](https://our.internmc.facebook.com/intern/diff/D31514095)

[ghstack-poisoned]

priyaramani committed:
Update on "Use kernel_func_name from aotCompiler"


Right now, assembly code generated for the a given method from the model is named wrapper or func by default. The function name is then replaced with a proper kernel_func_name after target specific assembly is generated. 
This PR propagates a desired kernel_func_name right from aotCompiler API so that the generated function has the needed name that doesn't need to be replaced later.

Differential Revision: [D31514095](https://our.internmc.facebook.com/intern/diff/D31514095)

[ghstack-poisoned]

priyaramani committed:
Update on "Use kernel_func_name from aotCompiler"


Right now, assembly code generated for the a given method from the model is named wrapper or func by default. The function name is then replaced with a proper kernel_func_name after target specific assembly is generated. 
This PR propagates a desired kernel_func_name right from aotCompiler API so that the generated function has the needed name that doesn't need to be replaced later.

Differential Revision: [D31514095](https://our.internmc.facebook.com/intern/diff/D31514095)

[ghstack-poisoned]

navahgar reviewed (approved):
LGTM. 

Please check the failures.

priyaramani committed:
Update on "Use kernel_func_name from aotCompiler"


Right now, assembly code generated for the a given method from the model is named wrapper or func by default. The function name is then replaced with a proper kernel_func_name after target specific assembly is generated. 
This PR propagates a desired kernel_func_name right from aotCompiler API so that the generated function has the needed name that doesn't need to be replaced later.

Differential Revision: [D31514095](https://our.internmc.facebook.com/intern/diff/D31514095)

[ghstack-poisoned]

priyaramani committed:
Update on "Use kernel_func_name from aotCompiler"


Right now, assembly code generated for the a given method from the model is named wrapper or func by default. The function name is then replaced with a proper kernel_func_name after target specific assembly is generated. 
This PR propagates a desired kernel_func_name right from aotCompiler API so that the generated function has the needed name that doesn't need to be replaced later.

Differential Revision: [D31514095](https://our.internmc.facebook.com/intern/diff/D31514095)

[ghstack-poisoned]

ngimel commented:
Reverted, breaks `test_profiler` as can be clearly seen on the PR CI.
